### PR TITLE
Prevents libuv/E2BIG errors when a large NOTICE file exists (fixes #7783)

### LINKED ISSUE #5420: Getting "spawn E2BIG" error after upgrading to yarn v1.5.1
**What is the current behavior?**

After upgrading from v1.4.1 to v1.5.1, all `yarn` commands are now failing with the following message:

```
error An unexpected error occurred: "spawn E2BIG".
info If you think this is a bug, please open a bug report with the information provided in "/Users/ev.haus/Git/web-platform/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

The `yarn-error.log` shows the following tracestack.

```
Trace: 
  Error: spawn E2BIG
      at _errnoException (util.js:1003:13)
      at ChildProcess.spawn (internal/child_process.js:330:11)
      at Object.exports.spawn (child_process.js:499:9)
      at /usr/local/Cellar/yarn/1.5.1/libexec/lib/cli.js:30226:24
      at new Promise (<anonymous>)
      at new F (/usr/local/Cellar/yarn/1.5.1/libexec/lib/cli.js:23451:28)
      at /usr/local/Cellar/yarn/1.5.1/libexec/lib/cli.js:30225:12
      at run (/usr/local/Cellar/yarn/1.5.1/libexec/lib/cli.js:41582:7)
      at BlockingQueue.maybePushConcurrencyQueue (/usr/local/Cellar/yarn/1.5.1/libexec/lib/cli.js:41597:7)
      at BlockingQueue.shift (/usr/local/Cellar/yarn/1.5.1/libexec/lib/cli.js:41592:10)
```

Rolling back to an older version of yarn makes the error go away.

**If the current behavior is a bug, please provide the steps to reproduce.**

Any `yarn` script command that writes to disk seems to be returning this, such as `yarn install`.

**What is the expected behavior?**

Should not be failing.

**Environment**

- node v9.6.1
- yarn v1.5.1
- macOS 10.13.3

### LINKED ISSUE #7783: The size of the npm_package_noticeText env variable is too large for node.js
**Do you want to request a *feature* or report a *bug*?**
Report a bug

**What is the current behavior?**
Starting with Node 12.11, they use libuv to lookup environment variables.  Yarn creates a variable called `npm_package_noticeText` which contains the full contents of a package's NOTICE.txt file and can exceed the max size allowable by libuv.  This causes yarn commands to fail when packaging the [CMake Tools extension for VS Code](https://github.com/microsoft/vscode-cmake-tools).

**If the current behavior is a bug, please provide the steps to reproduce.**
**Repro steps:**
* **Prerequisites:**
  * Install node.js version 12.11 or newer ([download](https://nodejs.org/en/download/))
  * Install yarn ([download](https://yarnpkg.com/lang/en/docs/install/#windows-stable))
  * Run `npm install -g vsce`
* Create a folder where you can clone the extension repository (e.g. `c:\git`)
* In that directory run: `git clone https://github.com/microsoft/vscode-cmake-tools.git`
* Run: `cd vscode-cmake-tools`
* Run: `vsce package --yarn`

**What is the expected behavior?**
* vsce should create a cmake-tools-[version].vsix file in the `vscode-cmake-tools` folder

**Actual Behavior:**
* node crashes and the following output is displayed:
```
Executing prepublish script 'yarn run vscode:prepublish'...
ERROR  Command failed: yarn run vscode:prepublish
Assertion failed: len < MAX_ENV_VAR_LENGTH, file c:\ws\deps\uv\src\win\util.c, line 1476
error Command failed with exit code 3221226505.
```

**Please mention your node.js, yarn and operating system version.**
node.js >= 12.11
yarn = 1.21.1
OS = Windows 10 18362

### PR DESCRIPTION
**Pull request by dbjorge**
**Summary**

This PR fixes #7783 by following the example from #7419 that was used to fix #5420.

TLDR: it prevents yarn from erroring out anytime you have a >40kb NOTICE file at the root of your repo and try to invoke any run-script that involves `exec()`ing a new node process.

This change has some backcompat risk (it no longer sets the `npm_package_noticeText` env var that was previously available to run scripts); I don't see any references to that variable name in an all-github code search, so I think the risk is lower than the benefit.

**Test plan**

Similarly to #7419, this doesn't add any new test code because the existing tests already cover verifying each value in the `IGNORE_MANIFEST_KEYS` list.

I manually verified that it addresses #7783 by testing it against a small test yarn project with a 40kb notice.txt file:

```powershell
C:\repos\testyarnbug | 4 | 18:49:56 05/01
> ls


    Directory: C:\repos\testyarnbug

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d----            5/1/2020  6:05 PM                node_modules
-a---            5/1/2020  6:08 PM          40002 notice.txt
-a---            5/1/2020  6:06 PM            180 package.json
-a---            5/1/2020  6:05 PM           2105 yarn.lock


C:\repos\testyarnbug | 5 | 18:55:43 05/01
> type package.json
{
  "name": "test",
  "version": "1.0.0",
  "license": "MIT",
  "scripts": {
    "echo": "cross-env echo hi"
  },
  "devDependencies": {
    "cross-env": "^7.0.2"
  }
}

C:\repos\testyarnbug | 6 | 18:55:50 05/01
> yarn echo
yarn run v1.22.4
$ cross-env echo hi
Assertion failed: len < MAX_ENV_VAR_LENGTH, file c:\ws\deps\uv\src\win\util.c, line 1478
error Command failed with exit code 3221226505.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Exit code -1073740791. Command took 2.161414 seconds.

C:\repos\testyarnbug | 7 | 18:55:58 05/01
> node C:\repos\yarn\bin\yarn.js echo
yarn run v1.23.0-0
$ cross-env echo hi
"hi"
Done in 1.06s.

C:\repos\testyarnbug | 8 | 18:56:04 05/01
>
```

### CODE REVIEW

**Comment by dbjorge** [comment]
The "Yarn Acceptance Tests" failures are unrelated to this PR; the errors they're seeing are all:

```
##[warning]An image label with the label macOS 10.13 does not exist.
```

This is happening because Azure DevOps dropped support for hosted `macOS-10.13` agents on March 23rd 2020 (see https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops)

**Comment by dbjorge** [comment]
The ci/circleci failures on linux-node12 and linux-node13 also appear to be timeouts unrelated to this PR; the same timeout has occurred in the same 2 jobs in both of the other two most-recently-updated PRs (#7495 [12](https://circleci.com/gh/yarnpkg/yarn/37813)/[13](https://circleci.com/gh/yarnpkg/yarn/37802), #8066 [12](https://circleci.com/gh/yarnpkg/yarn/37745)/[13](https://circleci.com/gh/yarnpkg/yarn/37746))

**Comment by wwahammy** [comment]
Is there someone we can talk to about getting this merged and a new release for yarn? This is killing me.

**Comment by dbjorge** [comment]
@arcanis , it looks like you've merged the most-recently-merged PRs in this repo; are you the right person to ask to take a look at this?

**Comment by dbjorge** [comment]
Closing this for inactivity; it seems clear that it isn't going to be merged into yarn 1
